numba/typed/typedlist.py

```diff
@@ -220,6 +220,10 @@ def _typed(self):
         """
         return self._list_type is not None
 
+    @property
+    def dtype(self):
+        return self._list_type.dtype if self._typed else None
```

Comment:
I am not sure that `None` is a good way to signal the type here. There are also None-typed typed lists that simply contain lots of `None` values. You will need to make sure that this doesn't conflict. My suggestion would be to make a list and append a `None` to so that it is refined and then check what `dtype` would result in

```python
l = List()
l.append(None)
...
```

Reply:
It will be of type `none`.
```python
>>> from numba.typed import List
>>> a = List()
>>> a.append(None)
>>> a
ListType[none]([None])
>>> a.dtype
none
>>> type(a.dtype)
<class 'numba.core.types.misc.NoneType'>
```

Reply:
That sounds good.